devito/operator.py

```diff
@@ -188,6 +192,36 @@ def __init__(self, expressions, **kwargs):
 
     # Compilation
 
+    def _add_implicit(self, expressions):
+        """
+        Create and add to the expression tree any associated implicit
+        expressions.
+        """
+        updated_expressions = []
+        for e in expressions:
+            subdomain = e.subdomain
+            if bool(subdomain):
+                try:
+                    dat = e._subdomain._ie_dat
+                except AttributeError:
+                    dat = None
+                if bool(dat):
+                    dims = [d.root for d in e.free_symbols if isinstance(d, Dimension)]
+                    sub_dims = [d.root for d in subdomain.dimensions]
+                    dims = list(set(dims).symmetric_difference(set(sub_dims)))
+                    dims.append(subdomain._implicit_dimension)
+                    implicit_expressions = []
+                    for i in dat:
+                        eq = Eq(i['rhs'], i['lhs'], implicit_dims=dims)
+                        implicit_expressions.append(eq)
+                    for ie in implicit_expressions:
```

Comment:
hang on: why do you need this extra loop? can you not directly populate ``updated_expressions`` in the previous loop, instead of having an additional list such as ``implicit_expressions`` ?

Reply:
Indeed. I think I needed it before for some reason but after some changes it's no longer needed.